worker/uniter/resolver.go

```diff
@@ -58,7 +58,8 @@ func (s *uniterResolver) NextOp(
 	// waiting to be shutdown.
 	if localState.UpgradeSeriesPrepareStatus == model.UnitCompleted &&
 		remoteState.UpgradeSeriesPrepareStatus == model.UnitCompleted {
-		return nil, resolver.ErrNoOperation
+		logger.Criticalf("We hit here were we are supposed to idle when shutting down.")
```

Comment:
Did you mean to leave in this test logging?

Reply:
Yes, this logging should remain until the machine agent is ready to shut down the units. This messages should replaced with the code that causes the Uniter to idle, that is, returning a the no-op.